server/s3_delete_object_action.h

```diff
@@ -54,6 +54,7 @@ class S3DeleteObjectAction : public S3ObjectAction {
 
   // delete marker metadata
   std::shared_ptr<S3ObjectMetadata> delete_marker_metadata;
+  bool is_object_delete_marker;
```

Comment:
Is it necessary to keep this copy? It seems anywhere you are using this, `object_metadata->is_delete_marker()` would work just fine?

Reply:
it is needed 
after object metadata deleted we need to check in cleanup that is it delete marker or not
and during cleanup we can not use object_metadata->is_delete_marker()